discovery.md

```diff
@@ -691,108 +695,102 @@ Content-Type: application/json
-than once within a request.
+If other service attributes are included in the request, those SHOULD be
+ignored for the purposes of processing the request.
 
-The follow responses are defined by this specification:
-- `201 Created` if all the specified Services were processed successfully.
-  - If there was only one Service in the incoming request, then the response
-    MUST include an HTTP `Location` Header with a value that points to the
-    correspoding Service.
-  - If the request contained more than one Service, then the HTTP `Location`
-    Header MUST NOT appear in the response.
-  - The HTTP Response Body MUST include an array of `id` values whose order
-    and values MUST match the list of Services in the request.
+Service deletions MUST be processed in the order in which they are given. Thus,
+if a Service is specified multiple times the entire request MUST fail.
 
-Other responses are allowed, but not defined by this specification.
+The follow responses are defined by this specification:
+- `200 OK` if all the specified Services were deleted successfully.
+  - The HTTP Response Body MUST include an array of the Service values at the
+    time of deletion.  The order of those values MUST match that of the
+    Services in the request.
+- `400 Bad Request` if the first error encountered is a missing `id`
+- `404 Not Found` if the first error encountered is that there is no Service
+  with a given `id`
+- `409 Conflict` if the first error encountered is that a given epoch was less
+  than or equal to the existing epoch of the Service with a given `id`
 
 The format of the HTTP Request MUST adhere to the following:
 ```
 Content-Type: application/json
 
 [
   {
-    "url": "{url}",
-    "name": "{name}",
-    ... remainder of Service attributes ...
-  }
+    id: "{id}",      // REQUIRED
+    epoch: "{epoch}" // OPTIONAL
+  },
   ...
 ]
 ```
 
-The format of the HTTP Response MUST adhere to the following:
+The format of the `200 OK` HTTP Response MUST adhere to the following:
 ```
+200 OK
 Content-Type: application/json
 
 [
-  "id": "{id}"
+  {
+    "url": "{url}",
+    "name": "{name}",
+    ... remainder of Service attributes ...
+  }
   ...
 ]
 ```
 
-#### `PUT /services/{id}`
-
-This MUST update an existing Service in a Discovery Endpoint. The Body of the
-request message MUST contain a single Service definition. The Service MUST
-contain an `id` attribute that matches the `{id}` in the `PUT` URL.
+#### `POST /services/{id}`
 
-The request MAY include the `epoch` attribute. If present then the Discovery
-Endpoint MUST ensure that the value on the incoming request matches the
-current value on the Service. If they do not match then an error MUST be
-generated and the Service MUST NOT be updated. However, if the incoming
-request does not include this attribute then this verification MUST NOT
-be done.
+This MUST update the collection of Services in the Discovery Endpoint to
```

Comment:
collection? isn't it a singleton?

Reply:
The service is.  Yet the set of services known by the DE, while itself a kind of singleton, seems reasonably called a collection.  The next line is where the singleton nature of the service is reflected but perhaps the language is clumsy.  I'm open to alternative phrasings.  Something that I was intending to promote to the reader's consciousness is that the API is based on the collection of resources model of REST.  By upserting a singleton Service version here the set of all services is modified to contain that version replacing the old version if it exists.

Reply:
ok - rereading it, I can live with it.